### DISCUSSION TITLE
Is there example about how to use ParallelTrainer?

### DISCUSSION THREAD

**Discussion by namecq** [Toni-SM/skrl#152] [Q&A] [answered]
Hi.
  I cannot find any tutorials on using ParallelTrainer in the official documentation. Then I tried to modify the code in torch_gymnasium_penduluum_ppo.py to use ParallelTrainer (In win10), but encountered an error. 

`agent = PPO(models=models,
            memory=memory,
            cfg=cfg,
            observation_space=env.observation_space,
            action_space=env.action_space,
            device=device)

agents=[]
for _ in range(4):
    agents.append(agent)

#configure and instantiate the RL trainer
cfg_trainer = {"timesteps": 100000, "headless": True}
#trainer = SequentialTrainer(cfg=cfg_trainer, env=env, agents=[agent])
trainer = ParallelTrainer(env=env, agents=agents, cfg=cfg)

trainer.train()`

##########################################################
`RuntimeError: 
        An attempt has been made to start a new process before the
        current process has finished its bootstrapping phase.

        This probably means that you are not using fork to start your
        child processes and you have forgotten to use the proper idiom
        in the main module:

            if __name__ == '__main__':
                freeze_support()
                ...

        The "freeze_support()" line can be omitted if the program
        is not going to be frozen to produce an executable.`

**Comment by Toni-SM** [comment, marked as answer]
Hi @namecq 

Sorry for late response.
Under the section [Omniverse Isaac Gym environments (simultaneous learning by scope)](https://skrl.readthedocs.io/en/latest/intro/examples.html#omniverse-isaac-gym-environments-simultaneous-learning-by-scope) in skrl docs, there are several examples for the `ParallelTrainer`.
